### ISSUE TITLE
[Leo] Update the model selection header and info icon UI

### ISSUE THREAD

**Issue by aguscruiz** [brave/brave-browser#40435]
### Description

Design:
https://www.figma.com/design/m0Gdbf0wtqyfEFGm32VLLc/%F0%9F%94%84-Leo-AI--IN-PROGRESS-?node-id=21279-200775&t=xfKEkr7c1veLLmsb-1

We're going to remove the model header from the current design in favor of a new design
<img width="682" alt="image" src="https://github.com/user-attachments/assets/21f00034-fa98-4143-8875-a3cae302129f">

### New design
- Model introduction is removed
- Model used in answer is shown in the "Regenerate" menu on each answer
- When switching models during a conversation, a separator showing where the model changed and into which is shown
- Clicking the info icon on the separator shows more information about the model as a tooltip.

<img width="1933" height="1174" alt="Image" src="https://github.com/user-attachments/assets/7c7166c0-5b8c-4407-ac91-13ce8ab21697" />

**Comment by fallaciousreasoning** [comment]
So to Summarise:
1. We don't show the ModelInfo for the initial message if the model is `Automatic`
2. We show the ModelInfo before the response from the model where it was changed.

**Comment by aguscruiz** [comment]
Yes, correct. As soon as you select another mother from the list, the little banner should appear. 

Of course if you change the model 10 times before actually sending a prompt, we only show the latest change you made, not all, since there's no prompt being sent. No need to stack up all the changes made if there is no prompt being sent.

**Comment by LaurenWags** [comment]
Verified with

```
Brave | 1.95.96 Chromium: 152.0.7977.64 (Official Build) beta (arm64)
-- | --
Revision | 493f174f704408c8d7ed08d181be9af45a595ab0
OS | macOS Version 26.6.2 (Build 25G83)
```

Verified test plan from https://github.com/brave/brave-core/pull/37939#issue-4857548845

<details>
<summary>Side Panel - PASSED</summary>

- [✅ ] Open Leo AI and start a **new conversation** — no ModelIntro pill should appear for the model already selected on load
- [✅ ] Before sending a message, switch to a **different model** (e.g. Claude, Llama) — one pill appears at the **top** with the correct brand icon, model name, and info icon
- [✅ ] Before sending, switch to another model again — the top pill is **replaced** (still one pill), showing the latest model
- [✅ ] Switch **back to the original model** before sending — the pending pill is **removed**
- [✅ ] Hover/click the info icon — tooltip shows the model-specific intro message with the support link

https://github.com/user-attachments/assets/51d646f0-c1ff-47b2-8b10-6f593a6ec363

- [✅ ] Send the first message — the pill **stays at the top**; conversation continues below it
- [✅ ] Start (or continue) a conversation on a **specific model**, then switch to **Automatic** — an Automatic pill appears at the current position
- [✅ ] From that Automatic pill, switch back to the specific model before sending — the pending Automatic pill is **removed**

https://github.com/user-attachments/assets/2c0ebe7e-6fe5-4a9e-aeb4-6e5da3708274

- [✅ ] In an ongoing conversation, switch models and send a prompt — a pill appears below the latest reply; earlier pills stay put
- [✅ ] After sending, switch to another model (including Automatic) — a **new** pill appears at the new position; earlier pills remain
- [✅ ] Before sending the next prompt, switch models again — the pending inline pill is **replaced**; earlier pills stay put
- [✅ ] Switch to a different conversation / clear history — markers reset (session-only, expected)
- [✅ ] Reload the conversation — markers are gone (session-only, expected)

https://github.com/user-attachments/assets/c9ee262e-1bcb-4498-86a0-b49490bb333c

- [✅ ] Verify appearance in **light and dark** themes

Dark theme used to verify functionality above, spot checked on Light theme to confirm that the model pill displayed as expected, tooltip displayed as expected, etc.

https://github.com/user-attachments/assets/46f7bd6c-78c4-4f78-824e-3958bf433c3b

</details>

<details>
<summary>Full Page - PASSED</summary>

- [✅ ] Open Leo AI and start a **new conversation** — no ModelIntro pill should appear for the model already selected on load
- [✅ ] Before sending a message, switch to a **different model** (e.g. Claude, Llama) — one pill appears at the **top** with the correct brand icon, model name, and info icon
- [✅ ] Before sending, switch to another model again — the top pill is **replaced** (still one pill), showing the latest model
- [✅ ] Switch **back to the original model** before sending — the pending pill is **removed**
- [✅ ] Hover/click the info icon — tooltip shows the model-specific intro message with the support link

https://github.com/user-attachments/assets/67fb7817-c640-4378-95d1-28f1321a181e

- [✅ ] Send the first message — the pill **stays at the top**; conversation continues below it
- [✅ ] Start (or continue) a conversation on a **specific model**, then switch to **Automatic** — an Automatic pill appears at the current position
- [✅ ] From that Automatic pill, switch back to the specific model before sending — the pending Automatic pill is **removed**

https://github.com/user-attachments/assets/9f3182a1-9757-4d24-bd4e-4d5ea3a50368

- [✅ ] In an ongoing conversation, switch models and send a prompt — a pill appears below the latest reply; earlier pills stay put
- [✅ ] After sending, switch to another model (including Automatic) — a **new** pill appears at the new position; earlier pills remain
- [✅ ] Before sending the next prompt, switch models again — the pending inline pill is **replaced**; earlier pills stay put
- [✅ ] Switch to a different conversation / clear history — markers reset (session-only, expected)
- [✅ ] Reload the conversation — markers are gone (session-only, expected)

https://github.com/user-attachments/assets/22b461a4-8c49-49cd-b130-8f18b3908bf1

- [✅ ] Verify appearance in **light and dark** themes

Dark theme used to verify functionality above, spot checked on Light theme to confirm that the model pill displayed as expected, tooltip displayed as expected, etc.

https://github.com/user-attachments/assets/9ab3b57c-9f75-4c6d-9b17-5f51a8f53b00

</details>

**Comment by Uni-verse** [comment]
Verified on `Pixel 9a` using the following version: 

```
Brave	1.95.99 Chromium: 153.0.8010.28 (Official Build) (64-bit) 
Revision	9b376de4985feeec6a183b412af212244b5a8313
OS	Android 16; Build/CP1A.260505.005; 36; REL
```
Ensured the following based on the new design criteria in https://github.com/brave/brave-browser/issues/40435#issue-2464164809
- [x] verified that model introduction is removed.
- [x] verified that model used is shown in the "regenerate" menu.
- [x] verified that when swtiching models mid conversation, there is a separator shown where the model changed. 
- [x] verified that all labels and tooltips are shown, at the top of chat as well as when switching models.

example | example | example | example
--|--|--|--
<img width="653" height="1356" alt="Image" src="https://github.com/user-attachments/assets/0e688a7b-8c8e-4451-97a5-cce89a45d4f2" /> | <img width="653" height="1356" alt="Image" src="https://github.com/user-attachments/assets/2f7348ba-3fbc-4d60-bf91-dbf63e8f0035" /> | <img width="653" height="1356" alt="Image" src="https://github.com/user-attachments/assets/dd3c1dcc-3406-4e7c-9664-13892b153293" /> | <img width="653" height="1356" alt="Image" src="https://github.com/user-attachments/assets/3e62bc4f-cbea-4c61-b132-973110f8af35" />

**Comment by Uni-verse** [comment]
Verified on `iPhone 15` running `iOS 26.5` using version `1.95 (102)`

Ensured the following based on the new design criteria in https://github.com/brave/brave-browser/issues/40435#issue-2464164809
- [x] verified that model introduction is removed.
- [x] verified that model used is shown in the "regenerate" menu.
- [x] verified that when switching models mid conversation, there is a separator shown where the model changed. 
- [x] verified that all labels and tooltips are shown, at the top of chat as well as when switching models.

example | example | example
--|--|--
<img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/83c7dfed-a3ca-4248-97f6-77234fef738b" /> | <img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/a4a58083-f607-46b0-a771-9e8699634b69" /> | <img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/34de9bad-6d09-43cf-9411-ebd2326e4ec2" />
<img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/b0430f1c-de1d-48e3-92f5-6dc1d7294369" /> | <img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/f4ebe4cd-891e-4e3e-88a8-4e53234164c2" /> | <img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/f2a5e825-413b-4a95-8940-57459a052190" />
